pkg/challenges/challenges_test.go

```diff
@@ -177,6 +215,67 @@ func Test_isURISubjectAllowed(t *testing.T) {
 	}
 }
 
+func Test_isDomainAllowed(t *testing.T) {
+	tests := []struct {
+		name    string
+		subject string // Parsed to url.URL
+		issuer  string // Parsed to url.URL
+		want    error
+	}{{
+		name:    "match",
+		subject: "accounts.example.com",
+		issuer:  "accounts.example.com",
+		want:    nil,
+	}, {
+		name:    "issuer subdomain",
+		subject: "example.com",
+		issuer:  "accounts.example.com",
+		want:    nil,
+	}, {
+		name:    "subject subdomain",
+		subject: "profiles.example.com",
+		issuer:  "example.com",
+		want:    nil,
+	}, {
+		name:    "subdomain mismatch",
+		subject: "profiles.example.com",
+		issuer:  "accounts.example.com",
+		want:    nil,
+	}, {
+		name:    "subject domain too short",
+		subject: "example",
+		issuer:  "example.com",
+		want:    fmt.Errorf("subject URI hostname too short: example"),
+	}, {
+		name:    "issuer domain too short",
+		subject: "example.com",
+		issuer:  "issuer",
+		want:    fmt.Errorf("issuer URI hostname too short: issuer"),
+	}, {
+		name:    "domain mismatch",
```

Comment:
nit: maybe a test case for "domain mismatch, subdomain match"

Reply:
To me this is already covered by domain mismatch, because the function documents that it only checks top-level and second-level domains. The test for `subdomain mismatch` exists to show that subdomains are not checked.

Reply:
IMO the test is for triple-checking that we weren't lying when we wrote the documentation :)

I could imagine a sloppy implementation where that would fail (e.g., if you accidentally used index `0` instead of `len(..)-2`.

Reply:
Added another test!